test/parallel/test-abortsignal-timeout.js

```diff
@@ -0,0 +1,113 @@
+// Flags: --no-warnings --expose-gc --expose-internals
+'use strict';
+
+require('../common');
+const { aborted } = require('util');
+
+const {
+  ok,
+  notStrictEqual,
+  strictEqual,
+} = require('assert');
+
+const {
+  kWeakHandler,
+} = require('internal/event_target');
+
+const { setTimeout: sleep } = require('timers/promises');
+const { test } = require('node:test');
+
+test('Test AbortSignal timeout', async () => {
+  // Test AbortSignal timeout
+  const signal = AbortSignal.timeout(10);
+  ok(!signal.aborted);
+  await sleep(20);
+  ok(signal.aborted);
+  strictEqual(signal.reason.name, 'TimeoutError');
+  strictEqual(signal.reason.code, 23);
+});
+
+test("AbortSignal timeout doesn't prevent the signal from being garbage collected", async () => {
+  let ref;
+  {
+    ref = new globalThis.WeakRef(AbortSignal.timeout(1_200_000));
+  }
+
+  await sleep(10);
+  globalThis.gc();
+  strictEqual(ref.deref(), undefined);
+});
+
+test("AbortSignal with a timeout is not gc'd while there is an active listener on it", async () => {
+  // Test that an AbortSignal with a timeout is not gc'd while
+  // there is an active listener on it.
+  let ref;
+  function handler() {}
+  {
+    ref = new globalThis.WeakRef(AbortSignal.timeout(1_200_000));
+    ref.deref().addEventListener('abort', handler);
+  }
+
+  await sleep(10);
+  globalThis.gc();
+  notStrictEqual(ref.deref(), undefined);
+  ok(ref.deref() instanceof AbortSignal);
+
+  ref.deref().removeEventListener('abort', handler);
+
+  await sleep(10);
+  globalThis.gc();
+  strictEqual(ref.deref(), undefined);
+});
+
+test('If the event listener is weak, however, it should not prevent gc', async () => {
```

Comment:
I think that the current test name only makes sense with the context of the previous name, but if it failed in CI reading the name would not mean much, I suggest something like this:
```suggestion
test('AbortSignal with a timeout is gc'd while there is a weak listener active on it', async () => {
```
WDYT?

Reply:
thanks, rephrased it a little, take a look 😄